File: alg.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Security.Policy;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace alg
{
    internal class Program
    {
        static void Main()
        {
            1

            Console.WriteLine("Введите фамилию: ");
            string fam = Console.ReadLine();
            Console.WriteLine("Введите оценку: ");
            string mark = Console.ReadLine();

            Console.WriteLine($"Поздравляю, {fam}! У Вас автоматом {mark} за экзамен.");


            2
            Console.WriteLine("Введите количество чисел: ");
            int Num = Convert.ToInt32(Console.ReadLine());

            int[] ints = new int[Num];

            float sum = 0;
            int test;

            for (int i = 0; i < Num; i++)
            {
                Console.WriteLine($"Введите число номер {1 + i}");
                test = Convert.ToInt32(Console.ReadLine());
                ints[i] = test;
                sum += Convert.ToSingle(test);
            }

            Console.WriteLine($"среднее значение {sum / Num} округление {Math.Round(sum / Num)} в меньшую {Math.Floor(sum / Num)}");


            3


            bool cheakNum = false;
            bool cheakZagl = false;
            bool cheakProp = false;
            bool cheakSpecSym = false;



            while (true)
            {
                Console.WriteLine("Введите пароль: ");
                string password = Console.ReadLine();
                for (int i = 0; i < password.Length; i++)
                {

                    if (char.IsDigit(password[i]))
                    {
                        cheakNum = true;
                    }
                    if (char.IsUpper(password[i]))
                    {
                        cheakZagl = true;
                    }
                    if (char.IsLower(password[i]))
                    {
                        cheakProp = true;
                    }
                    if (!char.IsLetterOrDigit(password[i]))
                    {
                        cheakSpecSym = true;
                    }

                }
                if (password.Length < 8 || password.Length > 20)
                {
                    Console.WriteLine("Пароль от 8 до 20");
                }
                else if (!cheakNum)
                {
                    Console.WriteLine("В пароле должны быть числа");
                }
                else if (!cheakZagl)
                {
                    Console.WriteLine("В пароле должны быть заглавные буквы");
                }
                else if (!cheakProp)
                {
                    Console.WriteLine("В пароле должны быть прописные буквы");
                }
                else if (!cheakSpecSym)
                {
                    Console.WriteLine("В пароле должны быть спецсимволы");
                }
                else
                {
                    Console.WriteLine("пароль подходит");
                    break;
                }

            }


            4
            var random = new Random();
            int score = 0;

            for (int i = 0; i < 3; i++)
            {


                int a = random.Next(1, 24);
                int b = random.Next(1, 24);
                int c = random.Next(1, 24);

                //Console.WriteLine($"{a} {b} {c}");

                Console.WriteLine($"Введите число номер {1 + i}: ");
                int nums = Convert.ToInt32(Console.ReadLine());

                if (nums == a || nums == b || nums == c)
                {
                    score += 1;
                }




            }
            if (score >= 2)
            {
                Console.WriteLine("Вы победили ");
            }
            else
            {
                Console.WriteLine("Вы проиграли");
            }

            5
            Random random = new Random();
            DateTime rod = DateTime.Now.Date.AddDays(random.Next(-7, 1));
            double price = random.Next(10, 200);
            int srokGodn = random.Next(1, 7);

            Console.WriteLine($"дата изготовления: {rod}, цена: {price} руб., срок: {srokGodn} дня");

            if (rod == DateTime.Now.Date)
            {

            }
            else if (rod.AddDays(srokGodn) > DateTime.Now.Date)
            {
                price *= 0.8;
            }
            else
            {
                price = 0;
            }
            Console.WriteLine(price);



            6

            var person_dict = new Dictionary<string, List<object>>()
            {
                { "person_id", new List<object>() {41, 84, 72 } },
                { "person_name",  new List<object>(){"Keanu Reeves", "Jim Carrey", "Негр"} }
            };
            var travel_log = new Dictionary<string, List<object>>()
            {
                { "travel_id", new List<object>() { 101, 102, 105 } },
                { "person_id", new List<object>() { 41, 72, 41 } },
                { "country", new List<object>() { "USA", "Japan", "China" } }
            };

            Console.WriteLine("Кто вам интересен: ");
            string name = Console.ReadLine();

            int cheak = 0;
            int cheakCoutry = 0;

            for (int i = 0; i < person_dict["person_name"].Count; i++)
            {
                if (name == Convert.ToString(person_dict["person_name"][i]))
                {
                    for (int j = 0; j < travel_log["person_id"].Count; j++)
                    {
                        if (Convert.ToInt32(person_dict["person_id"][i]) == Convert.ToInt32(travel_log["person_id"][j]))
                        {
                            cheakCoutry = j;
                            cheak++;
                        }

                    }
                }
            }
            if (cheak == 2)
            {
                Console.WriteLine("Several");
            }
            else if (cheak == 1)
            {
                Console.WriteLine(travel_log["country"][cheakCoutry]);
            }
            else
            {
                Console.WriteLine("None");
            }
            7

            Random random = new Random();
            List<object> MarksMask = new List<object> { "Б", "П", "Н", 2, 3, 4, 5 };
            List<DateTime> dateMarks = new List<DateTime> { };
            List<object> Marks = new List<object> { };

            for (int i = 0; i < 11; i++)
            {
                Marks.Add(MarksMask[random.Next(MarksMask.Count)]);
                dateMarks.Add(DateTime.Now.AddDays(i).Date);


            }
            for (int j = 0; j < Marks.Count; j++)
            {
                Console.WriteLine($"{Marks[j]}|| {dateMarks[j]}");
            }

            Console.WriteLine("");
            int NumB = 0;
            for (int c = 0; c < Marks.Count; c++)
            {
                if (Convert.ToString(Marks[c]) == "Б" && dateMarks[c].Month == DateTime.Now.Month)
                {
                    NumB++;
                    Console.WriteLine(dateMarks[c]);
                }

            }
            Console.WriteLine($"Количество пропусков по болезни в этом месяце {NumB}");

            8
                var student = new Student();
            student.SetStudent("Островский", "Bc-21", "2023");
            student.Print(student);
            student.GenerateMarks();
            student.PrintMarks();

            var student2 = new Student();
            student2.SetStudent("Ytuhs", "Bc-21", "2023");
            student2.Print(student2);
            student2.GenerateMarks();
            student2.PrintMarks();
            var group = new Group();
            group.SetGroup("Ис-21");
            group.AddStudents(student);
            group.AddStudents(student2);
            group.PrintAllStudINGoup(group);




        }
        class Student
        {
            public List<object> marks = new List<object> { };
            public List<object> dateMarks = new List<object> { };
            public List<object> randomMaks = new List<object> { "б", "п", "н", 2, 3, 4, 5 };
            public string fio;
            public string year_of_admission;
            public string group;

            public void SetStudent(string name, string year_of_admission, string group)
            {

                this.fio = name;
                this.group = group;
                this.year_of_admission = year_of_admission;



            }
            public void Print(Student student)
            {
                Console.WriteLine(student.fio);
                Console.WriteLine(student.year_of_admission);
                Console.WriteLine(student.group);
            }
            public void GenerateMarks()
            {
                Random random = new Random();
                for (int i = 0; i < 11; i++)
                {
                    this.marks.Add(this.randomMaks[random.Next(this.randomMaks.Count)]);
                    this.dateMarks.Add(DateTime.Now.AddDays(i).Date);

                }
            }
            public void PrintMarks()
            {
                for (int j = 0; j < this.marks.Count; j++)
                {
                    Console.WriteLine($"Оценка ученика {this.fio} :{this.marks[j]} || {this.dateMarks[j]}");
                }
            }

        }
        class Group
        {
            public string name;
            public List<Student> students = new List<Student> { };

            public void SetGroup(string name)
            {
                this.name = name;
            }

            public void AddStudents(Student student)
            {
                this.students.Add(student);
            }

            public void PrintAllStudINGoup(Group group)
            {
                for (int a = 0; a < group.students.Count; a++)
                {
                    Console.WriteLine($"Группа {group.name} Студент {group.students[a].fio}");
                }
            }
        }
    }
}

File: alg.py
import math
import string
import random
import datetime
1
Fam = input("Введите вашу фаимилию: ")
Mark = input("Введите оценку: ")
print(f"Поздравляю, {Fam}! У Вас автоматом {Mark} за экзамен.")


2
a = int(input("Введите сколько чисел: "))
mas = []
sum  = 0.0
for i in range(a):
    mas.append(int(input(f"Введите число номер {i+1}:")))
    sum+=float(mas[i])
print(f"{mas} среднее знач. {round(sum/a,2)}  округленние {round(sum/a) } в меньшую {int(sum/a) }")


3
NumCheak = False
UPCheak = False
LOwCheak = False
SpecSymvolCheak = False
Specsimvol = string.punctuation
while True:
    password = input("Введите пароль: ")
    
    for i in range(len(password)):
        if password[i].isdigit():
            NumCheak = True
        if password[i].isupper():
            UPCheak = True    
        if password[i].islower():
            LOwCheak = True 
        if password[i] in Specsimvol :
            SpecSymvolCheak = True
    if len(password) >= 20 or len(password) <= 8:
        print("Пароль должен быть от 8 до 20 символов")
    elif NumCheak == False:
        print("В пароле должна быть хотя бы одна цифра")
    elif UPCheak == False:
        print("В пароле должна быть хотя бы одна заглавная буква")
    elif LOwCheak == False:
        print("В пароле должна быть хотя бы одна строчная буква")    
    elif SpecSymvolCheak == False:
        print("В пароле должен быть хотя бы один спецсимвол")   
    else:
        print("Пароль подходит")
        break

4

Marks = 0

for i in range(3):
    a = random.randint(1,24)
    b = random.randint(1,24)
    c = random.randint(1,24)
    print(f"{a} {b} {c}")
    ansv = int(input("Ведите число: "))
    
    if ansv == a or ansv == b or ansv == c:
        Marks = Marks + 1
    
    print(f"{a} {b} {c} \nВаше число: {ansv} \nКоличество ваших балов: {Marks}\n ")
    
if Marks >= 2:
    print("Вы выиграли")
else:
    print("Вы проиграли")

5
dataizg = datetime.date.today() + datetime.timedelta(days=random.randint(-7,0))
godnodo = dataizg+ datetime.timedelta(days=random.randint(1,7)) 
price = random.randint(10,100)

print(f"Дата изготовление {dataizg}  годно до {godnodo} сегодня {datetime.date.today()} цена {price}")

if dataizg == datetime.date.today():
    pass
elif godnodo > datetime.date.today():
    price*=0.8
else:
    price = 0
    
print(f"Цена после перерасчёта {round(price,2)}")

6
person_dict = { 'person_id': [41, 84, 72], 
                'person_name': ['Keanu Reeves', 'Jim Carrey', 'Negr']} 
travel_log = { 'travel_id': [101, 102, 105], 
                'person_id': [41, 72, 41], 
                'country': [ 'USA', 'Japan', 'China']}

name = input("Кто вам интересен: ")
dopCheak = 0 

for i in range(len(person_dict.get('person_name'))):
    if name == person_dict.get('person_name')[i]:
        cheack = 0  
        for j in range (len(travel_log.get('person_id'))):
            if travel_log.get('person_id')[j] == person_dict.get('person_id')[i]:
                dopCheak = j
                cheack +=1
                
                  
if cheack == 0:
    print(None)
elif cheack == 1:
    print(travel_log.get('country')[dopCheak])
elif cheack == 2:
    print("Several")

7


Estimation  = []
date  = []

randomMaks = ["б", "п","н", 2, 3, 4, 5]

for i in range(11):
    Estimation.append(randomMaks[random.randint(0,6)])
    date.append(datetime.date.today() + datetime.timedelta(days=i))
        
for i in range(len(Estimation)):
    print(f"{Estimation[i]}||{date[i]}")
NUmBol = 0
for i in range(len(Estimation)):
    if Estimation[i] == "б" and date[i].month == datetime.date.today().month:
        print(date[i])
        NUmBol +=1
print(f"Количество пропусков по болезни за месяц {NUmBol} ")

    
    
8


randomMaks = ["б", "п","н", 2, 3, 4, 5]

class Student:
    fio = ""
    group = ""
    yerPost = ""
    
    Estimation = []
    date = []
    def __str__(self):
        return f"студент {self.fio} группа {self.group} год поступления {self.yerPost}"
        
    def __init__(self,fio,group,yerPost ):
        self.fio = fio
        self.group = group
        self.yerPost = yerPost
        

    def GenMarks(self):
        for i in range(11):
            self.Estimation.append(randomMaks[random.randint(0,6)])
            self.date.append(datetime.date.today() + datetime.timedelta(days=i))
    def PrintMarks(self):
        for i in range(len(Student.Estimation)):
            print(f"студент {self.name} оценки{Student.Estimation[i]}||{Student.date[i]}")
            
aboba1 = Student("негр1", "Ис-21", "18-06-2023")
print(aboba1)
aboba1.GenMarks()
aboba1.PrintMarks()

aboba2 = Student("негр2", "Ис-21", "18-06-2023")
print(aboba2)
aboba2.GenMarks()
aboba2.PrintMarks()

aboba3 = Student("негр3", "Ис-21", "18-06-2023")
print(aboba3)
aboba3.GenMarks()
aboba3.PrintMarks()


class Group:
    nameGroup = ""
    students = []
    def __init__(self,nameGroup):
        self.nameGroup = nameGroup
      
    def Print(self):
        for stud in self.students:
            print(f"студент группы {self.nameGroup} {stud}")
        
    
    def append(self, student):
        self.students.append(student)
        
abobaGroup = Group("Ис-21")
abobaGroup.append(aboba1)
abobaGroup.append(aboba2)
abobaGroup.Print()
        
